polynote-runtime/src/main/scala/polynote/runtime/python/PythonObject.scala

```diff
@@ -122,6 +140,33 @@ object PythonObject {
 
   trait Runner {
     def run[T](task: => T): T
+    def asScalaList(obj: PythonObject): List[Any]
+    def asScalaMap(obj: PythonObject): Map[Any, Any]
   }
 
+}
+
+/**
+  * A [[PythonObject]] which is refined by a constant literal string type indicating its type name. Python objects
+  * will be assigned a type like this (i.e. `TypedPythonObject["DataFrame"]`) so that we can use its python type to
+  * find specific reprs for that type. For example, we could make an instance of type:
+  *
+  *     ReprsOf[ TypedPythonObject[Witness.`"DataFrame"`.T] ]
+  *
+  * which would be selected for a PythonObject["DataFrame"]. Then, this instance could (for example) return a similar
+  * streaming data representation to what the Scala DataFrame instance does, enabling the built-in data viz.
+  */
```

Comment:
LMK if above needs more explanation... constant types are kind of an odd thing to grasp.